cyral/resource_cyral_sidecar_listener.go

```diff
@@ -402,5 +428,34 @@ func getSidecarListenerSchema() map[string]*schema.Schema {
 				},
 			},
 		},
+		SQLServerSettingsKey: {
+			Description: "SQL Server settings.",
+			Type:        schema.TypeSet,
+			Optional:    true,
+			// Notice the MaxItems: 1 here. This ensures that the user can only specify one this block.
+			MaxItems:      1,
+			ConflictsWith: []string{S3SettingsKey, MySQLSettingsKey, DynamoDbSettingsKey},
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					SQLServerVersionKey: {
+						Description: "Advertised SQL Server version. Required (and only relevant) for " +
+							"Listeners of type 'sqlserver' " +
+							"The format of the version should be <major>.<minor>.<build_number> " +
+							"API will validate that the version is a valid version number. " +
+							"Major version is an integer in range 0-255. " +
+							"Minor version is an integer in range 0-255. " +
+							"Build number is an integer in range 0-65535. " +
+							"Example: 16.0.1000 " +
+							"To get the version of the SQL Server runtime, run the following query: " +
+							"SELECT SERVERPROPERTY('productversion') " +
+							"Note: If the query returns a four part version number, only the first three parts " +
+							"should be used. Example: 16.0.1000.6 -> 16.0.1000",
+						Type:     schema.TypeString,
+						Optional: false,
```

Comment:
```suggestion
```

Reply:
In API Sqlserver settings is optional, if provided though, it has to carry the version so I believe we want to align with that here.

Reply:
Sure, my intention here is just to point out that there's no need to specify `Optional: false` - it can be omitted from the schema declaration, especially because we are already defining `Required: true`.

Reply:
ah, sorry I didn't get that at first :) Want me to remove and push or leave it be?

Reply:
@gengdahlCyral thats just a nit, so up to you! Let me know once we can merge the PR and I can help with that, changes look good to me.

Reply:
Cool, I removed this line, it's a bit confusing to have optional and required mixed (and not really valid when I read the docs). 
Feel free to merge. @VictorGFM